src/project-manager.ts

```diff
@@ -483,13 +483,18 @@ export class ProjectManager implements Disposable {
 		}
 		if (!rootdirs.has('') && !this.configs.has('')) {
 			// collecting all the files in workspace by making fake configuration object
-			this.configs.set('', new ProjectConfiguration(this.localFs, '/', this.versions, '', {
+			const tsConfig: any = {
 				compilerOptions: {
 					module: ts.ModuleKind.CommonJS,
 					allowNonTsExtensions: false,
 					allowJs: true
 				}
-			}, this.traceModuleResolution, this.logger));
+			};
+			// if there is at least one config, giving no files to default one
```

Comment:
Agree with 1) and agree that we should stay alive (we shouldn't worry about "staying alive" == process crash, we have solid error handling at the RPC layer), but does the empty configuration in the second case actually provide any features? Or would it be equivalent to just bailing out because no configuration was found for the file?

What about the case where you have a tsconfig in e.g. `./src/tsconfig.json`, but also a gulpfile.js at the root? Can we still provide IntelliSense for the gulpfile.js?